src/js/modules/RoutingApp.js

```diff
@@ -25,195 +26,143 @@ import {
-                    session ? (
-                      <Redirect to="/myaccount/profile" />
-                    ) : (
-                      <SignUpPage />
-                    )}
+                  path={section.permalink}
+                  key={`section${section.id}`}
+                  render={props => <SectionPage section={section} />}
                 />
+              );
+            })}
+            {allSections.map(section => {
+              return (
                 <Route
                   exact
-                  path="/myaccount/profile"
-                  key={"profile"}
-                  render={() =>
-                    session ? <ProfilePage /> : <Redirect to="/myaccount" />}
+                  path={`${section.permalink}/:article_slug`}
+                  key={`article${section.id}`}
+                  component={ArticlePage}
                 />
+              );
+            })}
+            {descriptions.map(description => {
+              return (
                 <Route
                   exact
-                  path="/myaccount/profile/edit"
-                  key={"editProfile"}
-                  render={() =>
-                    session ? (
-                      <EditProfilePage />
-                    ) : (
-                      <Redirect to="/myaccount" />
-                    )}
+                  path={`/about/${description.slug}`}
+                  key={`description${description.id}`}
+                  render={props => (
+                    <DescriptionPage description={description} />
+                  )}
                 />
-                <Route
-                  exact
-                  path={"/recommended"}
-                  key={"recommended"}
-                  component={RecommendedPage}
-                />
-                <Route
-                  exact
-                  path={"/latest"}
-                  key={"latest"}
-                  component={LatestPage}
-                />
-                <Route
-                  exact
-                  path={"/search"}
-                  key={"search"}
-                  component={SearchPage}
-                />
-                <Route
-                  path="/404-page-not-found"
-                  key={"notFound"}
-                  component={NotFoundPage}
-                />
-                <Route
-                  path="*"
-                  key={"404"}
-                  render={() => <Redirect to="/404-page-not-found" />}
-                />
-              </Switch>
-            </PageLayout>
-          )}
-          {initDataError && (
-            <DataErrorPage error={initDataError} action={this.prepareData} />
-          )}
-        </div>
-      </ConnectedRouter>
-    );
-  }
-}
+              );
+            })}
+            <Route
+              exact
+              path={"/contributors/:contributor_slug"}
+              key={"contributors"}
+              render={props => <ContributorPage match={props.match} />}

```

Comment:
idk why you have to do this. Route automatically gives you match

Reply:
Just learned that that is the case if you use `component={...}`. But if you use `render=` and manually inject props, the `Route`'s props are not passed down.